torchx/test/torchx/nx_test.exs

```diff
@@ -249,42 +250,58 @@ defmodule Torchx.NxTest do
 
   describe "Nx.as_type" do
     test "basic conversions" do
```

Comment:
I think we should revert the ones under this describe because the equality will ensure the type is also equal, while assert_equal glosses over that